pkgs/tools/admin/gam/default.nix

```
@@ -0,0 +1,66 @@
+{ lib
+, fetchFromGitHub
+, python3
+}:
+
+python3.pkgs.buildPythonApplication rec {
+  pname = "gam";
+  version = "6.21";
+  format = "other";
+
+  src = fetchFromGitHub {
+    owner = "GAM-team";
+    repo = "gam";
+    rev = "v${version}";
+    sha256 = "sha256-MNLjdn0YCZ2nasMyuRM72VcGMTbr6jIQyj29+MV1OjQ=";
+  };
+
+  sourceRoot = "source/src";
+
+  patches = [
+    # Also disables update check
+    ./signal_files_as_env_vars.patch
+  ];
+
+  propagatedBuildInputs = with python3.pkgs; [
+    dateutil
+    distro
+    filelock
+    google-api-python-client
+    google-auth
+    google-auth-oauthlib
+    passlib
+    pathvalidate
+    setuptools
+  ];
+
+  # Use XDG-ish dirs for configuration. Maybe more than these needed.
+  makeWrapperArgs = [
+      "--set-default GAMUSERCONFIGDIR '~/.config/gam'"
+      "--set-default GAMCACHEDIR '~/.cache/gam'"
+      "--set-default GAMDRIVEDIR '~/'"
+    ];
```

Maybe something like this could work?

```suggestion
  makeWrapperArgs = [
    "--set-default GAMUSERCONFIGDIR '''${XDG_CONFIG_HOME:-~/.config}/gam'"
    "--set-default GAMCACHEDIR '''${XDG_CACHE_HOME:-~/.cache}/gam'"
    "--set-default GAMDRIVEDIR '''${HOME:-~}'"
  ];
```

---

I tried that. But `makeWapper` evaluates shell variables at build time and single quotes the vars in the wrapper, thus they wouldn't get expanded. I may consider patching the source to read the XDG env vars, but would like to avoid that.

---

Got it working with `--run`.